### ISSUE TITLE
Site Health PHP Default Timezone

### ISSUE THREAD

**Issue by jveals24** [chrisblakley/Nebula#2090]
I am getting an error in SiteHealth when this theme is enabled -- 
PHP default timezone is invalid
Performance
PHP default timezone was changed after WordPress loading by a date_default_timezone_set() function call. This interferes with correct calculations of dates and times.

**Comment by chrisblakley** [comment]
Yeah this is actually something I've struggled with a bit– and maybe someone out there can shed some light here.

This is the function/code that the Site Health Check has a problem with:
```php
add_action('init', array($this, 'set_default_timezone'), 1);
public function set_default_timezone(){
	$timezone_option = get_option('timezone_string');
	if ( empty($timezone_option) ){
		$timezone_option = 'America/New_York';
	}

	//Note: this is disabled here but also at the hook above too!
	date_default_timezone_set($timezone_option);
}
```

This uses the timezone set via WordPress's options– and I would love to not use that, but when I remove this function from Nebula, the timestamps on posts revert to GMT– even when the WordPress timezone option is set.

I have several `@todo` comments in the code about this, but I have not been able to find a suitable resolution to this problem.

Thank you for making this issue– I will keep it open for any future progress on this and hopefully we can get it resolved!

**Comment by jveals24** [comment]
Have you also found that if you schedule a post, the schedule doesn't work properly? Trying to make sure I'm attributing the woes correctly lol

**Comment by chrisblakley** [comment]
I don't personally use the scheduled posts feature myself, but I will absolutely include that in tests– thanks for bringing it up. Off the top of my head, I am not sure if WP CRON is affected by that function– I don't **think** it triggers the `init` hook, so it may be unaffected.
